python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -147,5 +152,17 @@ def build_model(self):
         if self.model_weights:
             model.set_weights(self.model_weights)
 
-        prediction = model(tensor).numpy().ravel()
-        return pd.DataFrame(prediction, columns=["predictions"])
+        result_tensor = model(tensor)
+        if (
+            tf.rank(result_tensor) < 2
+            or tf.rank(result_tensor) == 2
+            and result_tensor.shape[1] < 2
+        ):
+            return pd.DataFrame(result_tensor.numpy().ravel(), columns=["predictions"])
+        else:
+            # TODO: remove when dataset side supports ndarray.
+            if result_tensor.shape[0] > 1:
+                return tf.unstack(result_tensor)
+            else:
+                return [result_tensor[0]]
```

Comment:
@clarkzinzow This is pretty ugly to me...
Maybe [this](https://github.com/ray-project/ray/issues/24208) will help remove some of the code here...